substrate/frame/conviction-voting/src/traits.rs

```diff
@@ -0,0 +1,39 @@
+use crate::AccountVote;
+use frame_support::dispatch::DispatchResult;
+
+pub trait VotingHooks<AccountId, Index, Balance> {
+	// Called when vote is executed.
+	fn on_vote(who: &AccountId, ref_index: Index, vote: AccountVote<Balance>) -> DispatchResult;
+
+	// Called when removed vote is executed.
+	// is_finished indicates the state of the referendum = None if referendum is cancelled, Some(true) if referendum is ongoing and Some(false) when finished.
+	fn on_remove_vote(who: &AccountId, ref_index: Index, ongoing: Option<bool>);
+
+	// Called when removed vote is executed and voter lost the direction to possibly lock some balance.
+	// Can Return the amount that should be locked for the conviction time.
+	fn locked_if_unsuccessful_vote(who: &AccountId, ref_index: Index) -> Option<Balance>;
```

Comment:
mby `balance_locked_on_unsuccessful_vote` would be better name or `always_locked_balance`